utils/contract-utils.js

```diff
@@ -14,6 +14,32 @@ const isValidContract = function (contract) {
 	return true;
 };
 
+const isValidLibrary = function (libraries) {
+	if (JSON.stringify(libraries) !== JSON.stringify({})) {
+		return true;
+	}
+
+	return false;
+}
+
+const isValidBytecode = function (libraries, bytecode) {
+	if (isValidLibrary(libraries)) {
+		for (const key in Object.keys(libraries)) {
+			if (libraries.hasOwnProperty(key)) {
```

Comment:
Can you reverse this logic as:

```
if (!libraries.hasOwnProperty(key)) {
continue;
}
// Rest of your logic
```

You will greatly reduce complexity like this